stdlib/public/core/UnicodeScalar.swift

```diff
@@ -258,6 +258,17 @@ extension UnicodeScalar : CustomStringConvertible, CustomDebugStringConvertible
   }
 }
 
+extension UnicodeScalar : LosslessStringConvertible {
+  public init?(_ description: String) {
+    if let v = UInt32(description) where (v < 0xD800 || v > 0xDFFF)
+      && v <= 0x10FFFF {
+      self = UnicodeScalar(v)
```

Comment:
I think the intent of the `LosslessStringConvertible` protocol is that `T(someInstanceOfT.description) == someInstanceOfT`.  That is, this initializer should parse back what `.description` returns.

```swift
(swift) var c: UnicodeScalar = "a"
// c : UnicodeScalar = "a"
(swift) c.description
// r0 : String = "\"a\""
```